### ISSUE TITLE
Auto-formatting redundant curly braces in specific string template fails to compile

### ISSUE THREAD

**Issue by axemorgan** [pinterest/ktlint#883]
## Expected Behavior
After formatting the following code (which *does* compile), the result should also compile.
Input (Bug.kt):
```kotlin
fun main() {
    println("This is: ${null}")
}
```
ktlint output:
```kotlin
fun main() {
    println("This is: $null")
}
```

## Observed Behavior
Compilation of the formatted code fails with this error: 
> Bug.kt:2:25: error: keyword cannot be used as a reference

## Steps to Reproduce
1. Format the input file using `ktlint -F Bug.kt`
2. Compile it: `kotlinc Bug.kt`
3. Compilation fails

## Your Environment
* ktlint: 0.37.2
* kotlinc: kotlinc-jvm 1.4.0 (JRE 13.0.1+9)

Note: I don't think it's particularly useful to use `${null}` in a string template, I just ran across this while formatting a large project. 😅

**Comment by romtsn** [comment]
I'm wondering if there are any other keywords that can be used in the string template?

**Comment by axemorgan** [comment]
I tried a few keywords from [this list](https://kotlinlang.org/docs/reference/keyword-reference.html), and the same problem occurs for `true` and `false`.  Those two and `null` seem to be the only keywords I can find that are valid expressions by themselves, otherwise "${keyword}" isn't valid code.

**Comment by romtsn** [comment]
Thanks for checking 👍
